src/translators/sql_translator.py

```diff
@@ -230,13 +281,40 @@ def _insert_entities_of_type(self,
             entries.append(values)
 
         # Insert entities data
-        p1 = table_name
-        p2 = ', '.join(['"{}"'.format(c.lower()) for c in col_names])
-        p3 = ','.join(['?'] * len(col_names))
-        stmt = "insert into {} ({}) values ({})".format(p1, p2, p3)
-        self.cursor.executemany(stmt, entries)
+        self._insert_entity_rows(table_name, col_names, entries, entities)
         return self.cursor
 
+    def _insert_entity_rows(self, table_name: str, col_names: List[str],
+                            rows: List[List], entities: List[dict]):
+        col_list = ', '.join(['"{}"'.format(c.lower()) for c in col_names])
+        placeholders = ','.join(['?'] * len(col_names))
+        stmt = f"insert into {table_name} ({col_list}) values ({placeholders})"
+        try:
```

Comment:
so this should be the "safe" way, right?

Reply:
if sql injection proof is what you mean, yes in principle the question marks should stop a an attack based on dodgy values---well as long as the driver we use and the server do their job right. but i didn't actually change the way it used to work, just factor that bit of code out into a separate method without even thinking about sql injection to be honest. now that i look at it with malicious eyes, this line could still be a problem actually

    col_list = ', '.join(['"{}"'.format(c.lower()) for c in col_names])

the "safe" stuff i had in mind for this PR was to stop as much as reasonably possible the second insert from bombing out---i.e. the insert we do to recover from a SQL failure in the existing one. so e.g. we only save the bare min and in the plainest possible way, see code below.